tests/foreman/ui/test_hostgroup.py

```diff
@@ -145,48 +209,77 @@ def test_positive_update(self):
 
     @run_only_on('sat')
     @tier1
-    def test_positive_create_with_oscap_capsule(self):
-        """Create new hostgroup with oscap capsule
+    def test_positive_update_with_removed_arch(self):
+        """Create new hostgroup with content view, architecture and os
+        assigned. Then remove these attributes from the hostgroup. Attempt to
+        update hostgroup with the same architecture
 
-        :id: c0ab1148-93ff-41d3-93c3-2ff139349884
+        :id: 431d88a3-590e-42cf-b45d-09b1d8a62b30
 
-        :expectedresults: Hostgroup is created with oscap capsule
+        :expectedresults: Hostgroup is updated and no error is raised
+
+        :BZ: 1372917
 
         :CaseImportance: Critical
         """
         name = gen_string('alpha')
+        arch = entities.Architecture().create()
+        custom_os = entities.OperatingSystem(
+            architecture=[arch]).create()
         with Session(self.browser) as session:
             make_hostgroup(
-                    session,
-                    content_source=self.sat6_hostname,
-                    name=name,
-                    puppet_ca=self.sat6_hostname,
-                    puppet_master=self.sat6_hostname,
-                    oscap_capsule=self.sat6_hostname,
-                )
-            self.assertIsNotNone(self.hostgroup.search(name))
+                session,
+                name=name,
+                org=self.organization.name,
+                parameters_list=[
+                    ['Host Group', 'Lifecycle Environment', self.env.name],
+                    ['Host Group', 'Content View', self.cv.name],
+                    ['Operating System', 'Architecture', arch.name],
+                    [
+                        'Operating System',
+                        'Operating system',
+                        '{0} {1}'.format(custom_os.name, custom_os.major)
+                    ],
+                ],
+            )
+            self.hostgroup.search_and_click(name)
+            self.hostgroup.click(
+                tab_locators['hostgroup.tab_operating_system'])
+            self.assertEqual(
+                self.hostgroup.wait_until_element(
+                    locators['hostgroups.architecture_value']).text,
+                arch.name
+            )
+            self.hostgroup.click(locators['hostgroups.operating_system_clear'])
```

Comment:
Have you thought about some kind of update() logic like 'if parameter_value is None, then click 'clear' locator for parameter_name'? Or it was withdrawn as it won't be universal? To clear text fields we can pass empty string '', for such fields - None. Just a quick idea without deep analysis :)